src/reducers/InputOutput.js

```diff
@@ -46,15 +69,39 @@ export function materialsRemove(state, action) {
 }
 
 export function materialsExport(state, action) {
+    // TODO: GENERALIZE INDEX
```

Comment:
I plan to move the code for dealing with the basis to its own function so that it is not duplicated between export and save. It's just easier for me to see what's happening during testing if the pieces are contained in their function.